django/contrib/gis/db/models/functions.py

```diff
@@ -421,6 +422,36 @@ class Intersection(OracleToleranceMixin, GeomOutputGeoFunc):
     geom_param_pos = (0, 1)
 
 
+@BaseSpatialField.register_lookup
+class GeometryType(GeoFuncMixin, Transform):
+    function = "GeometryType"
+    output_field = CharField()
+    lookup_name = "geom_type"
+
+    def as_oracle(self, compiler, connection, **extra_context):
+        lhs, params = compiler.compile(self.lhs)
+
+        sql = f"""
+            CASE MOD({lhs}.SDO_GTYPE, 10)
```

Comment:
I'm pretty sure this won't work on Oracle, but I don't have any advice here.

Reply:
I'm not sure, I mentioned Sarah because  I don't know anyone here but if you can answer my question that would be valuable, thale CI has no option to test with oracle right? I will need to do it manually on my machine?

Reply:
If it can be of any help I've been using this branch to get Oracle running locally against the test suite https://github.com/django/django-docker-box/issues/4#issuecomment-2104905724